libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -292,7 +292,26 @@ proc send*(p: PubSubPeer, msg: RPCMsg, anonymize: bool) {.raises: [].} =
     sendMetrics(msg)
     encodeRpcMsg(msg, anonymize)
 
-  asyncSpawn p.sendEncoded(encoded)
+  var sentMessages: seq[RPCMsg]
+  # Check if the encoded message size exceeds the maxMessageSize
+  if encoded.len > p.maxMessageSize:
+    var controlSent = false
+    # Split the RPCMsg into individual messages and send them separately
```

Comment:
this feels like it should be a responsibility of the caller - otherwise, we should change the `send` function tot take a list of `message` instead of a full `RPCMsg` - also, a single message might also be bigger than the limit so the utility of this change seems limited compared to the complexity it brings.

Reply:
I agree. I only did it here to use the encoded message and not add the overhead of calculating the message size. 

If individual messages bigger than the size limit are a common case, she should revisit the message limit. This change allows us to send messages below the size limit that can be sent individually but not as a group.

Reply:
Now we can split the messages before calling `send`. But then we need to create a `seq` to pass it. Wdyt @arnetheduck ?